### PR TITLE
Make CloserRule use guava's Closer

### PR DESCRIPTION
**Pull request by drcrallen**
Use Guava's closer instead of some custom hack job of mine.

### CODE REVIEW

File: processing/src/test/java/io/druid/segment/CloserRuleTest.java [hunk 1]
```diff
@@ -68,10 +65,10 @@ public void testAutoCloses() throws Throwable
     final CloserRule closer = new CloserRule(false);
     final AtomicBoolean closed = new AtomicBoolean(false);
     closer.closeLater(
-        new AutoCloseable()
+        new Closeable()
```
Comment (himanshug) [inline processing/src/test/java/io/druid/segment/CloserRuleTest.java]: now what is the difference between testCloses() and testAutoCloses() ?

**Comment by himanshug** [comment]
:+1: after https://github.com/druid-io/druid/pull/2579/files#r54838142 is resolved

File: processing/src/test/java/io/druid/segment/CloserRuleTest.java (hunk 1 above)
Comment (drcrallen) [inline processing/src/test/java/io/druid/segment/CloserRuleTest.java]: probably none. I tried not to tweak the tests too much. I'll delete it or modify to make more obvious.

Comment (drcrallen) [inline processing/src/test/java/io/druid/segment/CloserRuleTest.java]: fixed

**Comment by fjy** [comment]
:+1:

**Pull request merged**